tests/trees-adopted.test.ts

```diff
@@ -0,0 +1,132 @@
+import {
+	supabaseAnonClient,
+	supabaseServiceRoleClient,
+} from "../src/supabase-client";
+import { createTwoUsers, deleteUsers } from "./helper";
+
+describe("trees_adopted", () => {
+	let userAdoptId1: number | undefined;
+
+	let users: { userId1: string; userId2: string } = {
+		userId1: "",
+		userId2: "",
+	};
+
+	beforeAll(async () => {
+		users = await createTwoUsers();
+	});
+
+	afterAll(async () => {
+		await deleteUsers([users.userId1, users.userId2]);
+		const { error: deleteError } = await supabaseServiceRoleClient
+			.from("trees_adopted")
+			.delete()
+			.neq("id", -1);
+		if (deleteError) {
+			throw new Error("Failed to delete trees_adopted");
+		}
+	});
+
+	it("should not be able to adopt a tree if not authenticated", async () => {
+		const { data: adopt1, error: adoptError1 } = await supabaseAnonClient
+			.from("trees_adopted")
+			.insert({
+				uuid: users.userId1,
+				tree_id: "_0epuygrgg",
+			})
+			.select("*");
+		expect(adoptError1).toBeDefined();
+		expect(adopt1).toBeNull();
+	});
+
+	it("should be able to adopt a tree", async () => {
+		const { data, error } = await supabaseAnonClient.auth.signInWithPassword({
+			email: "user1@test.com",
+			password: "password1",
+		});
+		expect(data).toBeDefined();
+		expect(error).toBeNull();
+
+		const { data: adopt1, error: adoptError1 } = await supabaseServiceRoleClient
```

Comment:
why do we use the `serviceRoleClient` here? Can't we use the `supabaseAnonClient` (as we log-in just before?)

Reply:
yes, good catch. fixed it